sentry-failure/src/lib.rs

```diff
@@ -75,18 +103,20 @@ pub fn exception_from_single_fail<F: Fail + ?Sized>(
 
 /// Helper function to create an event from a `failure::Error`.
 pub fn event_from_error(err: &failure::Error) -> Event<'static> {
-    let mut exceptions = vec![];
-
-    for (idx, cause) in err.iter_chain().enumerate() {
-        let bt = match cause.backtrace() {
-            Some(bt) => Some(bt),
-            None if idx == 0 => Some(err.backtrace()),
-            None => None,
-        };
-        exceptions.push(exception_from_single_fail(cause, bt));
-    }
-
+    let mut exceptions: Vec<_> = err
+        .iter_chain()
+        .enumerate()
+        .map(|(idx, cause)| {
+            let bt = match cause.backtrace() {
+                Some(bt) => Some(bt),
+                None if idx == 0 => Some(err.backtrace()),
+                None => None,
+            };
+            exception_from_single_fail(cause, bt)
+        })
+        .collect();
```

Comment:
IMHO it was more readable before, but either way is fine.

Reply:
I was trying to avoid the temporary `Vec`, but looks like the source iterator is not exact-sized, so it has no `.rev()` :-(